numba/cuda/compiler.py

```diff
@@ -479,8 +481,11 @@ def __init__(self, py_func, argtypes, link=None, debug=False, inline=False,
         }
 
         tgt_ctx = cres.target_context
+        filename = cres.type_annotation.filename
+        linenum = int(cres.type_annotation.linenum)
```

Comment:
Also, the line number being a string in the type annotation is a minor problem, as LLVM expects line numbers to be integers (e.g. `5`), not strings (`"5"`).

Reply:
`foo.py_func.__code__.co_firstlineno` might help, as might `inspect` module?